File: P1/Main.cs
using System;
using System.Collections.Generic;

class MainClass {
    static void Main(string[] args) {
        // Get input from the user
        Console.Write("Enter single digit numbers separated by commas: ");
        string userInput = Console.ReadLine();

        // Extract single digit numbers into dynamic array
        List<int> numbers = new List<int>();
        for (int i = 0; i < userInput.Length; i++) {
            char c = userInput[i];
            if (c != ',') {
                numbers.Add(c - '0');
            }
        }

        // Print the list of numbers
        Console.Write("Your numbers are: [");
        for (int i = 0; i < numbers.Count; i++) {
            Console.Write(numbers[i]);
            if (i < numbers.Count - 1) {
                Console.Write(", ");
            }
        }
        Console.WriteLine("]");

        // Find the minimum and average manually
        int minVal = numbers[0];
        double totalSum = 0;
        for (int i = 0; i < numbers.Count; i++) {
            int num = numbers[i];
            if (num < minVal) {
                minVal = num;
            }
            totalSum += num;
        }
        double avg = totalSum / numbers.Count;

        // Output the minimum and average
        Console.WriteLine("Minimum number is: " + minVal);
        Console.WriteLine("Average is: " + avg);
    }
}

File: P1/Main.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    // Get input from the user
    cout << "Enter single digit numbers separated by commas: ";
    string userInput;
    getline(cin, userInput);

    // Extract single digit numbers into dynamic array
    vector<int> numbers;
    for (size_t i = 0; i < userInput.length(); i++) {
        char c = userInput[i];
        if (c != ',') {
            numbers.push_back(c - '0');
        }
    }

    // Print the list of numbers
    cout << "Your numbers are: [";
    for (size_t i = 0; i < numbers.size(); i++) {
        cout << numbers[i];
        if (i < numbers.size() - 1) {
            cout << ", ";
        }
    }
    cout << "]" << endl;

    // Find the minimum and average manually
    int minVal = numbers[0];
    double totalSum = 0;
    for (int num : numbers) {
        if (num < minVal) {
            minVal = num;
        }
        totalSum += num;
    }
    double avg = totalSum / numbers.size();

    // Output the minimum and average
    cout << "Minimum number is: " << minVal << endl;
    cout << "Average is: " << avg << endl;

    return 0;
}